esmvalcore/_recipe.py

```diff
@@ -632,23 +665,54 @@ def _get_preprocessor_products(variables, profile, order, ancestor_products,
     return products
 
 
+def _remove_temporal_preprocs(profile):
+    """Remove all temporal operations on fx variables."""
+    fx_profile = profile
+    temporal_preprocs = [
+        'extract_season',
+        'extract_month',
+        'annual_mean',
+        'seasonal_mean',
+        'time_average',
+        'regrid_time',
+    ]
+    for temporal_preproc in temporal_preprocs:
+        if temporal_preproc in profile:
+            fx_profile[temporal_preproc] = False
+
+    return fx_profile
+
+
 def _get_single_preprocessor_task(variables,
                                   profile,
                                   config_user,
                                   name,
                                   ancestor_tasks=None):
-    """Create preprocessor tasks for a set of datasets."""
+    """Create preprocessor tasks for a set of datasets w/ special case fx."""
     if ancestor_tasks is None:
         ancestor_tasks = []
     order = _extract_preprocessor_order(profile)
     ancestor_products = [p for task in ancestor_tasks for p in task.products]
-    products = _get_preprocessor_products(
-        variables=variables,
-        profile=profile,
-        order=order,
-        ancestor_products=ancestor_products,
-        config_user=config_user,
-    )
+
+    if variables[0]['frequency'] != 'fx':
+        products = list(_get_preprocessor_products(
+            variables=variables,
+            profile=profile,
+            order=order,
+            ancestor_products=ancestor_products,
+            config_user=config_user,
+        ))
+    else:
+        fx_profile = _remove_temporal_preprocs(profile)
+        # not do time extraction; this is a default preproc
+        fx_profile['extract_time'] = False
```

Comment:
no, that's a default preproc that needs to be removed upstream

Reply:
What do you mean by upstream? Putting this in `_remove_temporal_preprocs` seems to produce the exact same state, no?

Reply:
should be but last time I did it it ddin't work because `profile` gets populated with the default options only at the very last step before creating products, I meant downstream not upstream, my bad